src/biogeochem/CNCStateUpdate1Mod.F90

```diff
@@ -206,70 +219,57 @@ subroutine CStateUpdate1( num_soilc, filter_soilc, num_soilp, filter_soilp, &
                c = filter_soilc(fc)
-               ! TODO(wjs, 2017-01-02) Should some portion or all of the following fluxes
-               ! be moved to the updates in CStateUpdateDynPatch?
-               do i = i_litr_min, i_litr_max
-                  cf_soil%decomp_cpools_sourcesink_col(c,j,i) = &
-                       cf_soil%FATES_c_to_litr_c_col(c,j,i) * dt
-               end do
+               !
+               ! State update without the matrix solution
+               !
+               if (.not. use_soil_matrixcn) then
+                  cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) = &
+                       cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) &
+                       - ( cf_soil%decomp_cascade_hr_vr_col(c,j,k) + cf_soil%decomp_cascade_ctransfer_vr_col(c,j,k)) *dt
+               end if !not use_soil_matrixcn 
             end do
          end do
-      endif
-         
-      if ( .not. use_fates_sp ) then !use_fates
-         ! litter and SOM HR fluxes
-         do k = 1, ndecomp_cascade_transitions
+      end do
+      do k = 1, ndecomp_cascade_transitions
+         if ( cascade_receiver_pool(k) /= 0 ) then  ! skip terminal transitions
             do j = 1,nlevdecomp
                do fc = 1,num_soilc
                   c = filter_soilc(fc)
                   !
                   ! State update without the matrix solution
                   !
                   if (.not. use_soil_matrixcn) then
-                     cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) = &
-                       cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) &
-                       - ( cf_soil%decomp_cascade_hr_vr_col(c,j,k) + cf_soil%decomp_cascade_ctransfer_vr_col(c,j,k)) *dt
-                  end if !not use_soil_matrixcn 
-               end do
-            end do
-         end do
-         do k = 1, ndecomp_cascade_transitions
-            if ( cascade_receiver_pool(k) /= 0 ) then  ! skip terminal transitions
-               do j = 1,nlevdecomp
-                  do fc = 1,num_soilc
-                     c = filter_soilc(fc)
-                     !
-                     ! State update without the matrix solution
-                     !
-                     if (.not. use_soil_matrixcn) then
-                        cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) = &
+                     cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) = &
                           cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) &
                           + cf_soil%decomp_cascade_ctransfer_vr_col(c,j,k)*dt
-                     end if !not use_soil_matrixcn
-                  end do
+                  end if !not use_soil_matrixcn
                end do
-            end if
-         end do
-      end if
+            end do
+         end if
+      end do
 
-    if (.not. use_fates) then    
```

Comment:
yes, exactly!  The idea is that this section is ready in the future if we have another module (like crop for instance) that wants to run along-side fates. Those patches will be allocated, and allowed to do there thing.  I am aware though of the point you made up in another thread about future proofing. In this example we are future proofing, but it could be making the code a little harder to interpret today.

Reply:
OK, I'm going to make an issue for the name change of the filter, readability will really be helped if we make that the same. But, that can be a future BFB tag.

Reply:
#2068

Reply:
#2066